src/librustdoc/html/sources.rs

```diff
@@ -50,8 +50,14 @@ struct LocalSourcesCollector<'a, 'tcx> {
     src_root: &'a Path,
 }
 
-fn is_real_and_local(span: clean::Span, sess: &Session) -> bool {
-    span.cnum(sess) == LOCAL_CRATE && span.filename(sess).is_real()
+fn is_real_and_local(span: clean::Span, sess: &Session) -> Option<RealFileName> {
```

Comment:
If it's not a bool function, why would it have an `is_` name?

Perhaps it should be called `filename_real_and_local` instead?